### ISSUE TITLE
beats-dashboards should not use dynamic typing

### ISSUE THREAD

**Issue by LeeDr** [elastic/packetbeat-dashboards#94]
There's have a problem when you load beats-dashboards BEFORE saving other items from Kibana.  I first noticed the problem when I tried to use the Kibana Timelion plugin and got an error;

```
{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"mapper [hits] cannot be changed
from type [long] to [int]"}],"type":"illegal_argument_exception","reason":"mapper [hits] cannot be changed
from type [long] to [int]"},"status":400}
```

This is because when beats-dashboards were loaded, hits was saved as a long, but Kibana and [all?] plugins use an 'int' for hits.
@rashidkpc please correct me if I got this wrong.
- Kibana _could_ set the types when it first starts up to avoid this, but Rashid says that's front end code that runs in the browser so would be hard to do in a very automated way.
- Kibana _could_ provide an API for loading all of this (future).
- beats-dashboards _could_ set "hits" to be "int"s (and statically type all other fields?).  
- The _**easiest thing**_ right now might be to document NOT to install the beats-dashboards until AFTER the user has done a few things in Kibana to set hits as int.  Like saving a Saved Search.

**Comment by LeeDr** [comment]
Confirmed.  I started Kibana before installing the beats-dashboards, (I had a .marvel-es-1-\* index pattern), I went to Discover tab and used that index pattern and saved the default search.  Then I loaded beats-dashboards, and now Timelion does work.

**Comment by monicasarbu** [comment]
@LeeDr Thank you for testing the beats dashboards. I think the problem appears because when you load the dashboards, the default search is overwritten to the one included in the beats-dashboards (that is logstash-*).

**Comment by tsg** [comment]
A possible solution seems to be to run something like this in the beats-dashboards loading script before loading the searches/viz/etc.:

``` sh
curl -XPUT 'http://localhost:9200/.kibana/_mapping/search' -d'{"search": {"properties": {"hits": {"type": "integer"}, "version": {"type": "integer"}}}}'
```

I think this is essentially your third option and seems to be enough to get Timelion running. @rashidkpc: are there any other fields besides `hits` and `version` that might trigger the same issue?

**Comment by tsg** [comment]
This should be fixed now in master and 1.2.0 branch. Thank you @LeeDr and @rashidkpc for finding and investigating this!

**Comment by LeeDr** [comment]
I've verified on my integration test system :-)
